internal/manager/manager.go

```diff
@@ -65,32 +62,40 @@ func Start(cfg config.Config) error {
 		return fmt.Errorf("cannot build runtime manager: %w", err)
 	}
 
-	err = sdk.RegisterGatewayClassController(mgr, gc.NewGatewayClassImplementation(cfg, eventCh))
-	if err != nil {
-		return fmt.Errorf("cannot register gatewayclass implementation: %w", err)
-	}
-	err = sdk.RegisterGatewayController(mgr, gw.NewGatewayImplementation(cfg, eventCh))
-	if err != nil {
-		return fmt.Errorf("cannot register gateway implementation: %w", err)
-	}
-	err = sdk.RegisterHTTPRouteController(mgr, hr.NewHTTPRouteImplementation(cfg, eventCh))
-	if err != nil {
-		return fmt.Errorf("cannot register httproute implementation: %w", err)
-	}
-	err = sdk.RegisterServiceController(mgr, svc.NewServiceImplementation(cfg, eventCh))
-	if err != nil {
-		return fmt.Errorf("cannot register service implementation: %w", err)
-	}
-	err = sdk.RegisterSecretController(mgr, secret.NewSecretImplementation(cfg, eventCh))
-	if err != nil {
-		return fmt.Errorf("cannot register secret implementation: %w", err)
+	controllerConfigs := []controllerConfig{
+		{
+			objectType:           &gatewayv1beta1.GatewayClass{},
+			namespacedNameFilter: createFilterForGatewayClass(cfg.GatewayClassName),
+		},
+		{
+			objectType: &gatewayv1beta1.Gateway{},
+		},
+		{
+			objectType: &gatewayv1beta1.HTTPRoute{},
+		},
+		{
+			objectType:     &apiv1.Service{},
+			k8sEventFilter: predicate.ServicePortsChangedPredicate{},
+		},
+		{
+			objectType: &apiv1.Secret{},
+		},
+		{
+			objectType:     &discoveryV1.EndpointSlice{},
+			k8sEventFilter: k8spredicate.GenerationChangedPredicate{},
+			fieldIndexes: map[string]client.IndexerFunc{
+				index.KubernetesServiceNameIndexField: index.ServiceNameIndexFunc,
```

Comment:
Would it be better to export a fieldIndexes map for EndpointSlices?

So: 
`fieldIndexes: EndpointSliceFieldIndexes`